libs/editor/egui_editor/src/input/mutation.rs

```diff
@@ -59,14 +58,13 @@ pub fn calc(
         }
         Modification::ToggleStyle { region, style } => {
             let cursor = region_to_cursor(region, current_cursor, buffer, galleys, bounds);
-            apply_style(
-                cursor,
-                style.clone(),
-                region_completely_styled(cursor, RenderStyle::Markdown(style.clone()), ast),
-                buffer,
-                ast,
-                &mut mutation,
-            );
+            let unapply = region_completely_styled(cursor, &style, ast);
+            if !unapply {
+                for conflict in conflicting_styles(cursor, &style, ast) {
+                    apply_style(cursor, conflict, true, buffer, ast, &mut mutation)
+                }
+            }
```

Comment:
there's now a concept of two styles 'conflicting'; when applying a style, any conflicting styles applied to the same region are removed